src/ucs/sys/string.c

```diff
@@ -163,4 +164,24 @@ char* ucs_strncpy_safe(char *dst, const char *src, size_t len)
     return dst;
 }
 
+char *ucs_strtrim(char *str)
+{
+    char *start, *end;
+
+    /* point 'p' at first non-space character */
+    start = str;
+    while (isspace(*start)) {
+        ++start;
+    }
 
+    if (*start) {
+        /* write '\0' after the last non-space character */
+        end = start + strlen(start) - 1;
+        while (isspace(*end)) {
+            --end;
+        }
+        *(end + 1) = '\0';
+    }
+
+    return start;
```

Comment:
this implementation works fine for local buffers only, or for buffers where original buffer is release separately.

what about shift trimmed string to begin of original buffer? it will allow to use same pointer for original and trimmed string (trim it inplace)

Reply:
yes, this is also an option, but for the current use-case it was not needed..